rts/lra/test/arquillian-extension/src/main/java/io/narayana/lra/arquillian/ArquillianParametrized.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2021, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package io.narayana.lra.arquillian;
+
+import org.jboss.arquillian.junit.Arquillian;
+import org.junit.Ignore;
+import org.junit.runner.Description;
+import org.junit.runner.notification.RunNotifier;
+import org.junit.runners.Parameterized;
+import org.junit.runners.ParentRunner;
+import org.junit.runners.model.FrameworkField;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.InitializationError;
+import java.lang.reflect.Field;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Parameterized Arquillian JUnit runner adding the possibility to run parameterized JUnit test cases ,
```

Comment:
"parameterized JUnit" test cases sounds good but what are they?
I guess it is something to do with simplifying the version conformance test suite, maybe.
It is not at all clear what the purpose of this class is.

Reply:
The parametrized JUnit tests are tests which runs several times with different parameters - e.g. something here https://www.tutorialspoint.com/junit/junit_parameterized_test.htm
This is an adjustment of the Arquillian runner to be capable to run in the similar manner. It's for a generic use. The point is to use it for versions right now but it can be used in different tests as well.